utils/ixdtf/src/parsers/records.rs

```diff
@@ -0,0 +1,185 @@
+    pub minute: u8,
+    /// A second value.
+    pub second: u8,
+    /// A nanosecond value representing all sub-second components.
+    pub nanosecond: u32,
+}
+
+/// A `TimeZoneAnnotation` that represents a parsed `TimeZoneRecord` and its critical flag.
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq)]
+pub struct TimeZoneAnnotation<'a> {
+    /// Critical flag for the `TimeZoneAnnotation`.
+    pub critical: bool,
+    /// The parsed `TimeZoneRecord` for the annotation.
+    pub tz: TimeZoneRecord<'a>,
+}
+
+/// Parsed `TimeZone` data, which can be either a UTC Offset value or IANA Time Zone Name value.
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq)]
+pub enum TimeZoneRecord<'a> {
+    /// TimeZoneIANAName
+    Name(&'a str),
+    /// TimeZoneOffset
+    Offset(UTCOffsetRecord),
+}
+
+/// The parsed sign value, representing whether its struct is positive or negative.
+#[repr(i8)]
+#[allow(clippy::exhaustive_enums)] // Sign can only be positive or negative.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Sign {
+    /// A negative value sign, representable as either -1 or false.
+    Negative = -1,
+    /// A positive value sign, representable as either 1 or true.
+    Positive = 1,
+}
+
+impl From<bool> for Sign {
+    fn from(value: bool) -> Self {
+        match value {
+            true => Self::Positive,
+            false => Self::Negative,
+        }
+    }
+}
+
+/// A full precision `UtcOffsetRecord`
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct UTCOffsetRecord {
+    /// The `Sign` value of the `UtcOffsetRecord`
+    pub sign: Sign,
+    /// The hour value of the `UtcOffsetRecord`
+    pub hour: u8,
+    /// The minute value of the `UtcOffsetRecord`.
+    pub minute: u8,
+    /// The second value of the `UtcOffsetRecord`.
+    pub second: u8,
+    /// Any nanosecond value of the `UTCOffsetRecord`
+    pub nanosecond: u32,
+}
+
+/// The resulting record of a `Duration` parse.
+#[non_exhaustive]
+#[cfg(feature = "duration")]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct DurationParseRecord {
+    /// Duration Sign
+    pub sign: Sign,
+    /// The `years` value.
+    pub years: u32,
+    /// The `months` value.
+    pub months: u32,
+    /// The `weeks` value.
+    pub weeks: u32,
+    /// The `days` value.
+    pub days: u32,
+    /// The `hours` value.
+    pub hours: u32,
+    /// The `minutes` value.
+    pub minutes: u32,
+    /// The `seconds` value.
+    pub seconds: u32,
+    /// Any fraction part of a duration.
+    pub fraction: Option<DurationFraction>,
+}
+
+/// An enum representing the fraction part of a duration.
+#[non_exhaustive]
+#[cfg(feature = "duration")]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum DurationFraction {
+    /// The fraction value applied to an hour.
+    Hours(u64),
+    /// The fraction value applied to the minutes field.
+    Minutes(u64),
+    /// The fraction value applied to the seconds field.
+    Seconds(u32),
+}
```

Comment:
Thought: another way to model this, which may or may not be better, would be

```rust
pub enum TimeDuration {
    None,
    Hours {
        hours: u32,
        fraction: u64,
    },
    Minutes {
        hours: u32,
        minutes: u32,
        fraction: u64,
    },
    Seconds {
        hours: u32,
        minutes: u32,
        seconds: u32,
        fraction: u32,
    }
}
```

This more properly encodes the invariant that if hours have fractions, then minutes and seconds don't exist, for example.